tests/catalog/test_db.py

```diff
@@ -0,0 +1,120 @@
+
+    @pytest.fixture
+    def catalog(self, engine):
+        return DatabaseCatalog(engine)
+
+    def test_catalog_init(self, catalog):
+        assert isinstance(catalog.Session(), Session)
+
+    def test_catalog_touch_doesnt_exist(self, catalog):
+        parent_path = Path("/databases/postgresql/instances/localhost/schemas/some_db/tables/")
+        child_path = Path("some_table")
+
+        catalog.touch(parent_path / child_path)
+        assert catalog.ls(parent_path) == [str(child_path)]
+
+    def test_catalog_touch_does_exist(self, catalog):
+        parent_path = Path("/databases/postgresql/instances/localhost/schemas/some_db/tables/")
+        child_path = Path("some_table")
+
+        catalog.touch(parent_path / child_path)
+        assert catalog.ls(parent_path) == [str(child_path)]
+
+        catalog.touch(parent_path / child_path)
+        assert catalog.ls(parent_path) == [str(child_path)]
+
+    @pytest.mark.skip(reason="Waiting to rebase main")
+    def test_catalog_touch_deleted_path(self, catalog):
+        parent_path = Path("/databases/postgresql/instances/localhost/schemas/some_db/tables/")
+        child_path = Path("some_table")
+
+        catalog.touch(parent_path / child_path)
+        assert catalog.ls(parent_path) == ["localhost"]
+
+        catalog.rm(parent_path / child_path)
+        assert catalog.ls(parent_path) is None
+        assert catalog.children(parent_path) is None
+        assert catalog.read(parent_path) is None
+
+        catalog.touch(parent_path / child_path)
+        assert catalog.ls(parent_path) == ["localhost"]
+
+    def test_write(self, catalog):
+        parent_path = Path("/databases/postgresql/instances/localhost/schemas/some_db/tables")
+        child_path = Path("some_table")
+
+        metadata = Location(database="postgresql",
+                            instance="localhost",
+                            schema="some_db",
+                            table="some_table")
+
+        catalog.write(parent_path, metadata.dict(), patch=False)
+        assert catalog.read(parent_path) == metadata.dict()
+
+    def test_rm(self, catalog):
+        parent_path = Path("/databases/postgresql/instances/localhost/schemas/some_db/tables")
+        child_path = Path("some_table")
+        catalog.touch(parent_path / child_path)
+        assert catalog.ls(parent_path) == sorted([str(child_path)])
+
+        catalog.rm(parent_path / child_path)
+        assert catalog.ls(parent_path / child_path) is None
+
+    def test_ls_no_entry(self, catalog):
+        parent_path = Path("/databases/postgresql/instances/localhost/schemas/some_db/tables")
+        assert catalog.ls(parent_path) is None
+
+    def test_ls_one_entry(self, catalog):
+        parent_path = Path("/databases/postgresql/instances/localhost/schemas/some_db/tables")
+        child_path = Path("table_one")
+        catalog.write(parent_path / child_path, {})
+        assert sorted(catalog.ls(parent_path)) == sorted([str(child_path)])
```

Comment:
Curious why `sorted` here if one entry?

Reply:
It was just to have a consistent pattern when using `catalog.ls`. If you think that this is confusing, I can remove it.
It works without the `sorted`.